include/jwt-cpp/jwt.h

```diff
@@ -2668,6 +2668,162 @@ namespace jwt {
 			}
 		}
 	};
+	
+	template<typename json_traits>
+	class jwks {
+		using basic_claim_t = basic_claim<json_traits>;
+		template<typename> friend class jwks_keys;
```

Comment:
I am not convinced this is required

You could simply make the `protected` ctor public and remove this friend association